source/row_selector.ts

```diff
@@ -0,0 +1,316 @@
+import { ArrayTableModel } from './array_table_model';
+import { TableModel } from './table_model';
+import { AddRowOperation, MoveRowOperation, Operation }
+  from './operations';
+
+/** Provides the functionality needed to resize a table's columns. */
+export class RowSelector {
```

Comment:
Rename this to ```RowSelectionTableModel``` and have it extend ```TableModel```. This class is a ```TableModel``` and must implement the same interface as a ```TableModel```.